eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg

```diff
@@ -90,6 +90,19 @@ public interface Immutable<name>Set extends Immutable<name>Collection, <name>Set
         return this.reject(set::contains);
     }
 
+    /**
+     * Returns the set of all objects that are a member of exactly one of {@code this} and {@code set} (elements which
+     * are in one of the sets, but not in both). For instance, for the sets [1, 2, 3] and [2, 3, 4], the symmetric
+     * difference set is [1, 4] . It is the set difference of the union and the intersection.
+     *
+     * @since 11.0.
+     */
+    @Override
+    default Immutable<name>Set symmetricDifference(<name>Set set)
+    {
+        return this.toSet().symmetricDifference(set).toImmutable();
```

Comment:
Hi @prathasirisha, we might be able to optimize this slightly.

`return set.reject(this::contains, this.reject(set::contains, Sets.mutable.empty())).toImmutable();`

Reply:
`Sets` is not for primitives, no?

Reply:
Oh right... and we don't have this on primitive side yet.

Reply:
@mohrezaei, any other suggestions on the code for potential optimizations?

Reply:
@donraab nope. LGTM